internal/loggers/loggers.go

```diff
@@ -0,0 +1,90 @@
+/*
+ * @license
+ * Copyright 2023 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package loggers
+
+import (
+	"github.com/spf13/afero"
+	"io"
+	"strings"
+)
+
+// Logger defines the interface for a logging implementation
+type Logger interface {
```

Comment:
nit: We might want to add a compile time type assertion for zap and console logger? (var _ Logger = ...) 
Just to be safe for future extensions